src/dataflow/src/source/mod.rs

```diff
@@ -713,92 +765,113 @@ where
                     .operator_scheduled_counter
                     .inc();
 
-                while let Some(message) =
-                    source_info.get_next_message(&mut consistency_info, &activator)
-                {
-                    let partition = message.partition.clone();
-                    let offset = message.offset;
-
-                    // Update ingestion metrics
-                    // Entry is guaranteed to exist as it gets created when we initialise the partition.
-                    consistency_info
-                        .partition_metrics
-                        .get_mut(&partition)
-                        .unwrap()
-                        .offset_received
-                        .set(offset.offset);
-
-                    // Determine the timestamp to which we need to assign this message
-                    let ts = consistency_info.find_matching_timestamp(
-                        &id,
-                        &partition,
-                        offset,
-                        &timestamp_histories,
-                    );
-                    match ts {
-                        None => {
-                            // We have not yet decided on a timestamp for this message,
-                            // we need to buffer the message
-                            source_info.buffer_message(message);
+                let mut next_message = true;
+
+                while next_message {
```

Comment:
I think it would be more idiomatic to make this an infinite `loop` and break when necessary, rather than maintain the `next_message` flag.